tests/lora/test_lora_layers_old_backend.py

```diff
@@ -1522,13 +1533,16 @@ def test_lora_on_off(self, expected_max_diff=1e-3):
         with torch.no_grad():
             old_sample = model(**inputs_dict).sample
 
-        _, lora_params = create_unet_lora_layers(model)
+        _, lora_params = create_unet_lora_layers(model, rank=self.lora_rank)
         model.load_attn_procs(lora_params)
 
         with torch.no_grad():
             sample = model(**inputs_dict, cross_attention_kwargs={"scale": 0.0}).sample
 
-        model.set_default_attn_processor()
+        # Unload LoRA.
+        for module in model.modules():
+            if hasattr(module, "set_lora_layer"):
+                module.set_lora_layer(None)
```

Comment:
Otherwise, we will have to load the pipeline and then use `unload_lora_weights()` as told here:

https://github.com/huggingface/diffusers/blob/2e4dc3e25d6abaf5f15d7e5c62db75a8bd63f915/src/diffusers/models/attention_processor.py#L387

Reply:
We should probably add a `unload_lora()` function here

Reply:
Would prefer doing that in a follow up PR, okay?

Reply:
We should probably add a `unload_lora()` function here

Reply:
Yes sounds good!

Reply:
https://github.com/huggingface/diffusers/pull/6451